uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/Entitlement.php

```diff
@@ -79,4 +83,89 @@ private static function validateComment($params) {
     }
   }
 
+  /**
+   * Returns the calculated entitlement for the given JobContract,
+   * AbsencePeriod and AbsenceType
+   *
+   * @param \CRM_Hrjobcontract_BAO_HRJobContract $contract
+   * @param \CRM_HRLeaveAndAbsences_BAO_AbsencePeriod $period
+   * @param \CRM_HRLeaveAndAbsences_BAO_AbsenceType $type
+   *
+   * @return \CRM_HRLeaveAndAbsences_BAO_Entitlement|null If there's no entitlement for the given arguments, null will be returned
+   */
+  public static function getContractEntitlementForPeriod(
+    JobContract $contract,
+    AbsencePeriod $period,
+    AbsenceType $type
+  ) {
+    if(!$contract->id) {
+      throw new InvalidArgumentException("The given JobContract doesn't have an ID");
+    }
+
+    if(!$period->id) {
+      throw new InvalidArgumentException("The given AbsencePeriod doesn't have an ID");
+    }
+
+    if(!$type->id) {
+      throw new InvalidArgumentException("The given AbsenceType doesn't have an ID");
+    }
+
+    $entitlement = new CRM_HRLeaveAndAbsences_BAO_Entitlement();
```

Comment:
again here .. maybe get_class() is a better alternative rather than writing the full class name